modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -1504,6 +1504,23 @@ private void _validateEdge(
 				"Object relationship must not be between unmodifiable system " +
 					"object definitions to be an edge of a root context");
 		}
+
+		int objectDefinition1ObjectEntriesCount =
+			_objectEntryLocalService.getObjectEntriesCount(
+				objectDefinition1.getObjectDefinitionId());
+		int objectDefinition2ObjectEntriesCount =
+			_objectEntryLocalService.getObjectEntriesCount(
+				objectDefinition2.getObjectDefinitionId());
+
+		if ((objectDefinition1ObjectEntriesCount > 0) ||
+			(objectDefinition2ObjectEntriesCount > 0)) {
```

Comment:
> I thought we would check if the relationship field entries were filled out.

I opted to not include this business rule in this task, I created [another task](https://liferay.atlassian.net/browse/LPD-35169) for this use case to avoid a big pull request.